src/platform/telink/wifi/3.3.99.0/WiFiManager.cpp

```diff
@@ -179,6 +179,20 @@ CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanResultCallback resultCal
     mWiFiState          = WIFI_STATE_SCANNING;
     mSsidFound          = false;
 
+    if (!ssid.empty()) // Directed Scanning, Main Spec, 11.9.7.1. "ScanNetworks Command"
+    {
+        if (!mInternalScan) // don't erase in the middle of Connect procedure
+        {
+            Instance().mDirectedScanning = true;
+            mWantedNetwork.Erase();
+            memcpy(mWantedNetwork.ssid, ssid.data(), ssid.size());
```

Comment:
should we check mWantedNetwork.ssid having sufficient space for ssid.size()?

Reply:
On the controller's side we get `General error: 0x87 (CONSTRAINT_ERROR)` if SSID's size is above constraint but I still added the size check here just in case, thanks